Plugin.php

```diff
@@ -8,28 +8,192 @@
 namespace yii\composer;
 
 use Composer\Composer;
+use Composer\DependencyResolver\Operation\UpdateOperation;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\Installer\PackageEvent;
+use Composer\Installer\PackageEvents;
 use Composer\IO\IOInterface;
 use Composer\Plugin\PluginInterface;
+use Composer\Script;
+use Composer\Script\ScriptEvents;
 
 /**
  * Plugin is the composer plugin that registers the Yii composer installer.
  *
  * @author Qiang Xue <qiang.xue@gmail.com>
  * @since 2.0
  */
-class Plugin implements PluginInterface
+class Plugin implements PluginInterface, EventSubscriberInterface
 {
     /**
+     * @var array noted package updates.
+     */
+    private $_packageUpdates = [];
```

Comment:
having exceptions for this _ prefix is confusing. @klimov-paul, once you commented this prefix to be important to prevent collisions in overriding, so one expects this to be consistently implemented.